test-data/unit/check-functools.test

```diff
@@ -144,3 +144,99 @@ def f(d: D[C]) -> None:
 
 d: D[int]  # E: Type argument "int" of "D" must be a subtype of "C"
 [builtins fixtures/dict.pyi]
+
+[case testFunctoolsPartialBasic]
+from typing import Callable
+import functools
+
+def foo(a: int, b: str, c: int = 5) -> int: ...  # N: "foo" defined here
+
+p1 = functools.partial(foo)
+p1(1, "a", 3)  # OK
+p1(1, "a", c=3)  # OK
+p1(1, b="a", c=3)  # OK
```

Comment:
I added one of these, it's not very useful because it'll just show you `partial[return_type]`... would need a custom mypy type / some very involved generics logic / non-standard features to make it show up in the reveal

Reply:
Ah makes sense. The current approach seems fine then.